File: 07pair sum/pairSum.py
# brute force
def pair_sum(numbers, target_sum):
  for i in range(0, len(numbers)):
    for j in range(i+1, len(numbers)):
      if numbers[i] + numbers[j] == target_sum:
        return (i,j)

def better_pair_sum(numbers, target_sum):
  previous_nums = {}

  for index, num in enumerate(numbers):
    complement = target_sum-num

    if complement in previous_nums:
      return (previous_nums[comlpement], index)
  
    previous_nums[num] = index

File: Structy_2/2_Arrays_Strings/5_pairSum/pairSum.js
const pairSum = (numbers, targetSum) => {
  let previousNumbers = {};

  for (let i = 0; i < numbers.length; i++) {
    let num = numbers[i]
    let complement = targetSum - num
    if(complement in previousNumbers) 
    {return [i, previousNumbers[complement]]}
    previousNumbers[num] = i
    console.log(previousNumbers)
  }
};

console.log(pairSum([1, 2, 3], 5));
